blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+     *
+     * @param locators
+     * @param range
+     * @param gran
+     * @return
+     */
+    @Override
+    public Map<Locator, MetricData> getDatapointsForRange( List<Locator> locators, Range range, Granularity gran ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getReadTimerContext( "DBasicMetricsRW.getDatapointsForRange" );
+        try {
+
+            List<Locator> unknowns = new ArrayList<Locator>();
+            List<Locator> strings = new ArrayList<Locator>();
+            List<Locator> booleans = new ArrayList<Locator>();
+            List<Locator> numerics = new ArrayList<Locator>();
+
+            for ( Locator locator : locators ) {
+
+                Object type;
+                try {
+
+                    type = metadataCache.get( locator, DATA_TYPE_CACHE_KEY );
+
+                } catch ( CacheException e ) {
+
+                    LOG.error( "Unable to read cache", e );
+
+                    unknowns.add( locator );
+                    continue;
+                }
+
+
+                DataType metricType = new DataType( (String) type );
+
+                if ( type == null || !DataType.isKnownMetricType( metricType ) ) {
+
+                    unknowns.add( locator );
+                    continue;
+                } else if ( metricType.equals( DataType.STRING ) ) {
+
+                    strings.add( locator );
+                    continue;
+                } else if ( metricType.equals( DataType.BOOLEAN ) ) {
+
+                    booleans.add( locator );
+                    continue;
+                } else {
+
+                    numerics.add( locator );
+                }
+            }
+
+            Map<Locator, MetricData> metrics = new HashMap<Locator, MetricData>();
+
+            String columnFamily = CassandraModel.getBasicColumnFamilyName( gran );
+
+            metrics.putAll( super.getDatapointsForRange( numerics, range, columnFamily, gran ) );
+            metrics.putAll( getBooleanDataForRange( booleans, range ) );
+            metrics.putAll( getStringDataForRange( strings, range ) );
+            metrics.putAll( getNumericOrStringDataForRange( unknowns, range, columnFamily, gran ) );
+
+            return metrics;
+        }
+        finally {
+
+            ctx.stop();
+        }
+    }
+
+    /**
+     * Fetches a {@link com.rackspacecloud.blueflood.types.Points} object for a
+     * particular locator and rollupType from the specified column family and
+     * range
+     *
+     * @param locator
+     * @param rollupType
+     * @param range
+     * @param columnFamilyName
+     * @return
+     */
+    @Override
+    public Points getDataToRollup(final Locator locator,
+                                  RollupType rollupType,
+                                  Range range,
+                                  String columnFamilyName) throws IOException, CacheException {
+
+        if( columnFamilyName.equals ( CassandraModel.CF_METRICS_STRING_NAME ) ) {
```

Comment:
I'm ok leaving this check here but I know this will never happen. RollupRunnable is the only caller of this method and it would never try to roll String metrics.